x-pack/plugins/security_solution/public/management/hooks/use_with_show_responder.tsx

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback } from 'react';
+import type { ConsoleResponseActionCommands } from '../../../common/endpoint/service/response_actions/constants';
+import { isResponseActionSupported } from '../../../common/endpoint/service/response_actions/is_response_action_supported';
+import { HeaderSentinelOneInfo } from '../components/endpoint_responder/components/header_info/sentinel_one/header_sentinel_one_info';
+
+import { useUserPrivileges } from '../../common/components/user_privileges';
+import {
+  ActionLogButton,
+  getEndpointConsoleCommands,
+  OfflineCallout,
+} from '../components/endpoint_responder';
+import { useConsoleManager } from '../components/console';
+import { MissingEncryptionKeyCallout } from '../components/missing_encryption_key_callout';
+import type { SentinelOneAgentInfo } from '../../../common/types';
+import { RESPONDER_PAGE_TITLE } from './translations';
+import { getCommandKey } from '../components/endpoint_response_actions_list/components/hooks';
+
+type ShowResponseActionsConsole = (agentInfo: SentinelOneAgentInfo) => void;
```

Comment:
Maybe this answer my prior question: think this method is only for sentinel one.

Can we try to only have 1 hook for showing responder regarless of agent type? else, we're going to end up with many hooks - one for each 3rd party EDR.

Reply:
Yeah I'll update this to use a generic agent Info for non-endpoint agents!